buy_together_app/views.py

```diff
@@ -7,3 +9,16 @@ def main_page(request):
 
 def description_page(request):
     return render(request, 'buy_together_app/description.html')
+
+
+def log_in_page(request):
+    if request.method == 'POST':
+        user_name = request.POST.get('username')
+        user_pass = request.POST.get('password')
+        user = authenticate(request, username=user_name, password=user_pass)
+        if user is not None:
+            login(request, user)
+            return redirect('Main Page')
+        else:
+            messages.info(request, 'User name OR Password is incorrect')
```

Comment:
There is an option that the username does not exist

Reply:
if so the authenticate function will fail and user will be none, will be redirected back to login page.

Reply:
If the username is not existed we want to still in the same page and to tell the user that this username doesnot exist,look at my pr i did something similar

Reply:
`            messages.info(request, 'User name OR Password is incorrect')
    return render(request, 'buy_together_app/login.html')`
does exactly that🤷‍♂️